### ISSUE TITLE
GSNA:::lhm() and hence ( subnetSummary() ) incorrectly computes log harmonic means

### ISSUE THREAD

**Issue by JonathanUrbach** [JonathanUrbach/GSNA#16]
Significant bug:

subnetSummary() incorrectly computes log harmonic means. This is due to an incorrect function of `GSNA:::lhm()`. 

```
exp( GSNA:::lhm( log( c(1,2,3,4) ) ) )
```
> [1] 1.411765

whereas:
```
psych::harmonic.mean( c(1,2,3,4)  )
```
> [1] 1.92

**Comment by JonathanUrbach** [comment]
The `lse` function, implementing the "Log Sum of Exponents trick" for calculating the log of the sum of exponentiated arguments was changed to avoid numeric overruns that resulted in `Inf` values.


Old function:

```
lse <- function(a,b){ a + log( 1 + exp( b - a ) ) }
```

```
lse <- function(a,b){
  if( b - a > 709 ){ # Prevent numeric overrun
    b
  } else {
    a + log( 1 + exp( b - a ) )
  }
}
```

**Comment by JonathanUrbach** [comment]
The `lse()` function which implements the so-called "log sum of exponentials-trick" was causing numeric overruns thereby producing `Inf` values when the arguments differed by more than about 709. The solution was to change the `lse()` function.

```
#OLD lse() produces erroneous infinite values:
lse <- function(a,b){ a + log( 1 + exp( b - a ) ) }
```

```
# New lse() works
lse <- function(a,b){
  if( b - a > 709 ){ # Prevent numeric overrun
    b
  } else {
    a + log( 1 + exp( b - a ) )
  }
}
```

The fix will be incorporated into GSNA version 0.1.4.7.
